DataFormats/Phase2ITPixelCluster/interface/Phase2ITPixelCluster.h

```diff
@@ -0,0 +1,279 @@
+  int sizeY() const {verifyVersion(); return colSpan() +1;}
+  
+  
+  inline float charge() const {
+    float qm = 0.0;
+    int isize = thePixelADC.size();
+    for (int i=0; i<isize; ++i) 
+      qm += float(thePixelADC[i]);
+    return qm;
+  } // Return total cluster charge.
+  
+  inline uint32_t minPixelRow() const { return thePixelRow&MAXPOS;} // The min x index.
+  inline uint32_t maxPixelRow() const { verifyVersion(); return minPixelRow() + rowSpan();} // The max x index.
+  inline uint32_t minPixelCol() const { return thePixelCol&MAXPOS;} // The min y index.
+  inline uint32_t maxPixelCol() const { verifyVersion(); return minPixelCol() + colSpan();} // The max y index.
+  
+  
+  const std::vector<uint16_t> & pixelOffset() const { return thePixelOffset;}
+  const std::vector<uint32_t> & pixelADC() const { return thePixelADC;}
+  
+  // obsolete, use single pixel access below
+  const std::vector<Pixel> pixels() const {
+    std::vector<Pixel> oldPixVector;
+    int isize = thePixelADC.size();
+    oldPixVector.reserve(isize); 
+    for(int i=0; i<isize; ++i) {
+      oldPixVector.push_back(pixel(i));
+    }
+    return oldPixVector;
+  }
+  
+  // infinite faster than above...
+  Pixel pixel(int i) const {
+    return Pixel(minPixelRow() + thePixelOffset[i*2],
+		 minPixelCol() + thePixelOffset[i*2+1],
+		 thePixelADC[i]
+		 );
+  }
+  
+private:
+  
+  static int span_(uint32_t packed) { return packed >> POSBITS;}
+  static int overflow_(uint32_t packed) { return span_(packed)==uint32_t(MAXSPAN);}
+  static uint32_t pack_(uint32_t zmin, unsigned  int zspan) {
+    zspan = std::min(zspan, uint32_t(MAXSPAN));
+    return (zspan<<POSBITS) | zmin;
+  }
+public:
+  
+  int colSpan() const {return span_(thePixelCol); }
+  
+  int rowSpan() const { return span_(thePixelRow); }
+  
+  
+  bool overflowCol() const { return overflow_(thePixelCol); }
+  
+  bool overflowRow() const { return overflow_(thePixelRow); }
+  
+  bool overflow() const { return  overflowCol() || overflowRow(); }
+  
+  void packCol(uint32_t ymin, uint32_t yspan) {
+    thePixelCol = pack_(ymin,yspan);
+  }
+  void packRow(uint32_t xmin, uint32_t xspan) {
+    thePixelRow = pack_(xmin,xspan);
+  }
+  
+  
+  
+  /// mostly to be compatible for <610 
+  void verifyVersion() const {
+    if unlikely( thePixelRow<MAXPOS && thePixelCol<MAXPOS)
+		 const_cast<Phase2ITPixelCluster*>(this)->computeMax();
+  }
+  
+  /// moslty to be compatible for <610 
+  void computeMax()  {
+    int maxRow = 0;
+    int maxCol = 0;
+    int isize  = thePixelADC.size();
+    for (int i=0; i!=isize; ++i) {
+      int xsize  = thePixelOffset[i*2];
+      if (xsize > maxRow) maxRow = xsize;
+      int ysize = thePixelOffset[i*2+1] ;
+      if (ysize > maxCol) maxCol = ysize;
+    }
+    // assume minimum is correct
+    uint32_t minCol= minPixelCol();
+    packCol(minCol,maxCol);
+    uint32_t minRow= minPixelRow();
+    packRow(minRow,maxRow);
+  }
+  
+  // ggiurgiu@fnal.gov, 01/05/12 
```

Comment:
All the false history comments in all the files should be deleted. In the future, they could confuse readers about the age and origin of these files.